scripts/preconversion_assessment_script.py

```diff
@@ -197,27 +258,113 @@ def cleanup(required_files=None):
-    if required_files is None:
-        required_files = REQUIRED_FILES
-
+def verify_required_files_are_present(required_files):
+    """Verify if the required files are already present on the system."""
+    print("Checking if required files are present on the system.")
     for required_file in required_files:
         # Avoid race conditions
         try:
+            print("Checking for file %s" % required_file.path)
             with open(required_file.path, mode="r") as handler:
                 required_file.sha512_on_system = hashlib.sha512(handler.read())
                 required_file.is_file_present = True
         except (IOError, OSError):
             required_file.is_file_present = False
 
 
+def _generate_message_key(message, action_id):
+    """
+    Helper method to generate a key field in the message composed by action_id
+    and message_id.
+    """
+    new_message = copy.deepcopy(message)
+
+    new_message["key"] = "%s::%s" % (action_id, message["id"])
+    del new_message["id"]
+
+    return new_message
+
+
+def _generate_detail_block(message):
+    """
+    Helper method to generate the detail key that is composed by the
+    remediation and diagnosis fields.
+    """
+    new_message = copy.deepcopy(message)
+    detail_block = {
+        "remediation": {"context": ""},
+        "diagnosis": {"context": ""},
+    }
+
+    detail_block["remediation"]["context"] = new_message["remediation"]
+    detail_block["diagnosis"]["context"] = new_message["diagnosis"]
+
+    del new_message["remediation"]
+    del new_message["diagnosis"]
+
+    new_message["detail"] = detail_block
+    return new_message
+
+
+def _rename_dictionary_key(message, new_key, old_key):
+    """Helper method to rename keys in a flatten dictionary."""
+    new_message = copy.deepcopy(message)
+    new_message[new_key] = new_message.pop(old_key)
+    return new_message
+
+
+def apply_message_transform(message, action_id):
+    """Apply the necessary data transformation to the given messages."""
+    new_message = _generate_message_key(message, action_id)
+    new_message = _rename_dictionary_key(new_message, "severity", "level")
+    new_message = _rename_dictionary_key(new_message, "summary", "description")
+    new_message = _generate_detail_block(new_message)
+
+    # Appending the `modifiers` key to the message here for now. Once we have
+    # this feature in the frontend, we can populate the data with it.
+    new_message["modifiers"] = []
+
+    return new_message
+
+
+def transform_raw_data(raw_data):
+    """
+    Method that will transform the raw data given and output in the expected
+    format.
+    """
+    new_data = []
+    for action_id, result in raw_data["actions"].items():
+        # Format the results as a single list
+        for message in result["messages"]:
+            new_data.append(apply_message_transform(message, action_id))
+
+        new_data.append(apply_message_transform(result["result"], action_id))
+    return new_data
```

Comment:
So, I would not expect that the messages and actions are empty, as convert2rhel will always generate that even if they are just an empty dict/list. I can add a new test to check that even if we have empty dict/list we will not error out in this step. 

But one thing that I'm thinking right now is adding a new check to main.py to verify that we do have the json file to work with, otherwise, we just raise a ProcessError and exit the script.

Reply:
:thinking: don't know what you mean, the json report from analysis is present only after the command is run? If for some reason the json report file wouldn't be created then the exception message could be enough

Reply:
Yeah, it will be basically checking if the file/contents exists. If not, we should bail out.